pkg/gitprovider/gitlab/gitlab.go

```diff
@@ -156,6 +172,21 @@ func (p *provider) GetPullRequest(
 		return nil, fmt.Errorf("unexpected nil merge request")
 	}
 	pr := convertGitlabMR(glMR.BasicMergeRequest)
+	// Check if MR is in merge train when state is opened
+	if pr.Open {
+		mergeTrain, mtResp, err := p.mergeTrainClient.GetMergeRequestOnAMergeTrain(
+			p.projectName, int(id),
+		)
+		if err == nil && mtResp != nil && mtResp.StatusCode == 200 && mergeTrain != nil {
+			if mergeTrain.Status != "merged" && mergeTrain.Status != "skip_merged" {
+				pr.Queued = true
+			}
+		} else if mtResp != nil && mtResp.StatusCode == 403 {
```

Comment:
Is there a possibility that the `HTTP 403` error could conflict with someone providing invalid API credentials?